scripts/doc_coverage_model.py

```
@@ -0,0 +1,146 @@
+
+    Raises
+    ------
+    KeyError
+        If ``name`` is absent from ``entry``.
+    TypeError, ValueError
+        If the count cannot be an integer count, including JSON booleans, or
+        is negative.
+    """
+    count = entry[name]
+    if isinstance(count, bool):
+        raise ValueError("counts must be non-negative integers with with_docs <= total")
```

Ruff (0.16.2)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)